### PR TITLE
Improve integration tests stability

### PR DESCRIPTION
**Pull request by mwperina**
- Add delay to wait_for_unit_state method
- Wait for daemon-reload to finish

### CODE REVIEW

**Comment by dougsland** [comment]
LGTM but let me know when you need a final review (in case you add more patches)

**Comment by mwperina** [comment]
DO NOT MERGE - it will require to remove unnecessary changes after merging: https://github.com/containers/bluechi/pull/581

**Comment by mwperina** [comment]
> LGTM but let me know when you need a final review (in case you add more patches)

Ready for review :-;

File: tests/bluechi_test/container.py
Comment (mkemel) [inline tests/bluechi_test/container.py:137]: I think this functionality overlaps with what I did with `systemctl_start_and_wait` in #575 
Maybe `systemctl_start_and_wait` would be redundant after that

Comment (mwperina) [reply tests/bluechi_test/container.py:137]: When looking at our test code I was wondering if we could use `wait_for_unit_state_to_be` instead of hardcoded delay in `systemctl_start_and_wait`. I'm going to check it, but maybe we will need to implement `notify` service type functionality to our services: https://www.freedesktop.org/software/systemd/man/systemd.service.html#Options

Anyway at the moment this just improves existing code as we use `wait_for_unit_state_to_be`  much more often than `systemctl_start_and_wait`

Comment (mkemel) [reply tests/bluechi_test/container.py:137]: ack

**Review by mkemel — APPROVED** [review]

**Pull request merged**
